code/game/objects/items/devices/portable_chem_mixer.dm

```diff
@@ -148,33 +160,33 @@
 	if(!ui)
 		ui = new(user, src, ui_key, "PortableChemMixer", name, ui_x, ui_y, master_ui, state)
 		if(user.hallucinating())
-			ui.set_autoupdate(FALSE) //to not ruin the immersion by constantly changing the fake chemicals
+			ui.set_autoupdate(FALSE)	//to not ruin the immersion by constantly changing the fake chemicals
 		ui.open()
 
 /obj/item/storage/portable_chem_mixer/ui_data(mob/user)
-	var/list/data = list()
+	var/list/data = list()	///The data list that is sent to the tgui
 	data["amount"] = amount
 	data["isBeakerLoaded"] = beaker ? 1 : 0
 	data["beakerCurrentVolume"] = beaker ? beaker.reagents.total_volume : null
 	data["beakerMaxVolume"] = beaker ? beaker.volume : null
 	data["beakerTransferAmounts"] = beaker ? beaker.possible_transfer_amounts : null
-	var/chemicals[0]
-	var/is_hallucinating = user.hallucinating()
+	var/chemicals[0]	///Array of the chemicals in the source beakers
+	var/is_hallucinating = user.hallucinating()	///States if the user is currently hallucinating
 	if(user.hallucinating())
 		is_hallucinating = TRUE
 	for(var/re in dispensable_reagents)
-		var/value = dispensable_reagents[re]
-		var/datum/reagent/temp = GLOB.chemical_reagents_list[re]
+		var/value = dispensable_reagents[re]	///The individual dispensable reagents of the source beakers
+		var/datum/reagent/temp = GLOB.chemical_reagents_list[re]	///The reagents in the source beakers by chemical reagent list
 		if(temp)
-			var/chemname = temp.name
-			var/total_volume = 0
+			var/chemname = temp.name	///The name of the current reagent in question 
```

Comment:
To be clear, you only have to document the class level variables, variables defined inside a method are not interesting for documentation.

You should only put comments here if they are going to help the understanding of the process

Reply:
I apologize, I was not familiar with all the terminology at first (and still new to byond coding in general)
This time I only commented the class level variables and the two /proc/'s in the code, and I hope everything is correct now 😊